tests/src/test/scala/com/nvidia/spark/rapids/AdaptiveQueryExecSuite.scala

```diff
@@ -135,6 +135,7 @@ class AdaptiveQueryExecSuite
     val conf = new SparkConf()
         .set(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key, "true")
         .set(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key, "-1") // force shuffle exchange
+        .set(SQLConf.PARQUET_COMPRESSION.key, "uncompressed")
```

Comment:
Curious, why does compression need to be disabled for these tests?  I don't know of anyone ever doing this in practice, so it questions the utility of this config.  Do we need it to generate a dataset "large enough" to trigger AQE or something?  Would be good to put a comment explaining it, otherwise people will be tempted to remove it later.

Reply:
The test is failing on CI machine but passing locally.  @andygrove's comment in https://github.com/NVIDIA/spark-rapids/pull/694#issuecomment-689735910  was it could be either a) order of data being written to Parquet varies, and compresses differently so that we miss a threshold, or b) it is a race condition where the order of query stage completion matters.  To make the result deterministic we disabled the compression but it didn't help completely in this case as the latest build failed on 3.0.2.  There still seems to a race condition.

Reply:
The real issue here is that we are porting Spark tests that rely on the initial query plan believing that the data sets are large and therefore choosing SMJ over BHJ, and then optimizing for BHJ once query stages complete and show that the data sets are small (as in, below `AUTO_BROADCASTJOIN_THRESHOLD`).

These tests work in Spark because the test data is created using `spark.sparkContext.parallelize`, which creates instances of `ParallelCollectionRDD` which are then wrapped by `LogicalRDD` in the logical plan (and `RDDScanExec` in the physical plan). The `computeStats` method of `LogicalRDD` returns a row count of `Long.MaxValue`. Because we don't support `RDDScanExec` on the GPU, we had modified these tests to write the test data to Parquet, but this means that we get accurate stats during initial planning, and we cannot simply change the `AUTO_BROADCASTJOIN_THRESHOLD` config to get the desired plan, because it impacts other parts of the plan. We did think we had it working but given the inconsistencies between local runs and CI, I think we should remove the failing test from this PR and rethink how we can test this.

Reply:
@jlowe We have removed that failing test for now as adding the flag didn't help either.